src/ProtectedRoomsConfig.ts

```diff
@@ -0,0 +1,127 @@
+        const joinedRooms = await this.client.getJoinedRooms();
+        for (const roomRef of config.protectedRooms) {
+            const permalink = Permalinks.parseUrl(roomRef);
+            if (!permalink.roomIdOrAlias) continue;
+
+            let roomId = await this.client.resolveRoom(permalink.roomIdOrAlias);
+            if (!joinedRooms.includes(roomId)) {
+                roomId = await this.client.joinRoom(permalink.roomIdOrAlias, permalink.viaServers);
+            }
+            this.explicitlyProtectedRooms.add(roomId);
+        }
+    }
+
+    /**
+     * Load any rooms that have been explicitly protected from the account data of the mjolnir user.
+     * Will not ensure we can join all the rooms. This so mjolnir can continue to operate if bogus rooms have been persisted to the account data.
+     */
+    public async loadProtectedRoomsFromAccountData(): Promise<void> {
+        LogService.debug("ProtectedRoomsConfig", "Loading protected rooms...");
+        try {
+            const data: { rooms?: string[] } | null = await this.client.getAccountData(PROTECTED_ROOMS_EVENT_TYPE);
+            if (data && data['rooms']) {
+                for (const roomId of data['rooms']) {
+                    this.explicitlyProtectedRooms.add(roomId);
+                }
+            }
+        } catch (e) {
+            if (e.statusCode === 404) {
+                LogService.warn("ProtectedRoomsConfig", extractRequestError(e));
+            } else {
+                throw e;
+            }
+        }
+    }
+
+    /**
+     * Save the room as explicitly protected.
+     * @param roomId The room to persist as explicitly protected.
+     */
+    public async addProtectedRoom(roomId: string): Promise<void> {
+        this.explicitlyProtectedRooms.add(roomId);
+        await this.saveProtectedRoomsToAccountData();
+    }
+
+    /**
+     * Remove the room from the explicitly protected set of rooms.
+     * @param roomId The room that should no longer be persisted as protected.
+     */
+    public async removeProtectedRoom(roomId: string): Promise<void> {
+        this.explicitlyProtectedRooms.delete(roomId);
+        await this.saveProtectedRoomsToAccountData([roomId]);
+    }
+
+    /**
+     * Get the set of explicitly protected rooms.
+     * This will NOT be the complete set of protected rooms, if `config.protectAllJoinedRooms` is true and should never be treated as the complete set.
+     * @returns The rooms that are marked as explicitly protected in both the config and Mjolnir's account data.
+     */
+    public getExplicitlyProtectedRooms(): string[] {
+        return [...this.explicitlyProtectedRooms.keys()]
+    }
+
+    /**
+     * Persist the set of explicitly protected rooms to the client's account data.
+     * @param removeRooms Rooms that should be removed before saving the account data.
+     */
+    private async saveProtectedRoomsToAccountData(removeRooms: string[] = []): Promise<void> {
```

Comment:
Nit: The word "remove" makes it sound like you're doing some weird side-effect. Here, they're simply not saved, right? So, I'd rename `removeRooms` into `excludeRooms`, for instance and update the documentation to get replace "be removed" with "not be saved".

Reply:
That's a good point, but if these rooms were already persisted, then we would be removing them.

I have made the doc clearer and renamed.